File: Algorithms/1. Two Sum/swift/solution.one-pass.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var hashmap: [Int: Int] = [:]

        for (i, num) in nums.enumerated() {
            if let complement: Int = hashmap[target - num] {
                return [i, complement]
            }

            hashmap[num] = i
        }

        return []
    }
}

File: Algorithms/1. Two Sum/php/solution.one-pass.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target)
    {
        $hashmap = [];
        foreach ($nums as $i => $num) {
            $complement = $target - $num;
            if (array_key_exists($complement, $hashmap)) {
                return [$i, $hashmap[$complement]];
            }
            $hashmap[$num] = $i;
        }
    }
}
